controllers/csiscaleoperator_controller.go

```diff
@@ -577,45 +581,45 @@ func (r *CSIScaleOperatorReconciler) restartControllerPod(logger logr.Logger, in
 
 	logger.Info("restarting Controller Pod")
 	controllerPod := &corev1.Pod{}
-	controllerStatefulset, err := r.getControllerStatefulSet(instance)
+	controllerDeployment, err := r.getControllerDeployment(instance)
 	if err != nil {
-		logger.Error(err, "failed to get controller statefulset")
+		logger.Error(err, "failed to get controller deployment")
 		return err
 	}
 
 	logger.Info("controller requires restart",
-		"ReadyReplicas", controllerStatefulset.Status.ReadyReplicas,
-		"Replicas", controllerStatefulset.Status.Replicas)
+		"ReadyReplicas", controllerDeployment.Status.ReadyReplicas,
+		"Replicas", controllerDeployment.Status.Replicas)
 	logger.Info("restarting csi controller")
 
-	err = r.getControllerPod(controllerStatefulset, controllerPod)
+	err = r.getControllerPod(controllerDeployment, controllerPod)
 	if errors.IsNotFound(err) {
 		return nil
 	} else if err != nil {
 		logger.Error(err, "failed to get controller pod")
 		return err
 	}
 
-	return r.restartControllerPodfromStatefulSet(logger, controllerStatefulset, controllerPod)
+	return r.restartControllerPodfromDeployment(logger, controllerDeployment, controllerPod)
 }
 
-func (r *CSIScaleOperatorReconciler) getControllerPod(controllerStatefulset *appsv1.StatefulSet, controllerPod *corev1.Pod) error {
-	controllerPodName := fmt.Sprintf("%s-0", controllerStatefulset.Name)
+func (r *CSIScaleOperatorReconciler) getControllerPod(controllerDeployment *appsv1.Deployment, controllerPod *corev1.Pod) error {
+	controllerPodName := controllerDeployment.Name
```

Comment:
The podname consists of `<deploymentname>+<podhash>`, here I see only deployment name is passed, I am not sure if this works or not. Is the `Client.Get` request able to fetch correct pod details?

Reply:
Yeah, Thanks for the questions, while changing statfulset to deployment, I changed the type of this also to deployment.name, which should be pod.name actually. Added a fix for it.

Reply:
I saw the fix, `controllerPodName := controllerPod.Name`, here controllerPod is an empty object of type `corev1.Pod`. I doubt this would work. Is it working as expected?

Reply:
Yes, this is working as expected. I see operator pod restarting without any issue.